src/core/types.h

```diff
@@ -434,6 +434,7 @@ class Box {
     // Add a no-op constructor to make sure that we don't zero-initialize cls
     Box() {}
 
+    void* operator new(size_t size, BoxedClass* cls, size_t nitems) __attribute__((visibility("default")));
```

Comment:
DEFAULT_CLASS is basically just a wrapper around `Box::operator new()` that adds a default argument, and C++ classes are free to not use it and just inherit the base version (not sure of any examples that do this).  I think it should be safe to move this to BoxVar and then change DEFAULT_CLASS_VAR to call `BoxVar::operator new`?